### PR TITLE
Disable signal handler tests under ASAN.

### PR DESCRIPTION
**Pull request by dnoe**
Some specific ones already had to be disabled but other tests no longer
pass under the latest Clang ASAN because they mysteriously have no output.

The ENVOY_BUILD_SHA is updated to the latest so CI uses that CI image.

### CODE REVIEW

File: ci/ci_steps.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-ENVOY_BUILD_SHA=fc747b3c2fd49b1260484572071fe4194cd6824d
+ENVOY_BUILD_SHA=2867b337182c79ec5229a4261ecdadbf916f49ab4629dfea787cf76c91952dc4
```
Comment (mattklein123) [inline ci/ci_steps.sh]: I think this should be 1d2a2100708d1012cd61054ce04d4dc4e3b03ff2 ?

https://hub.docker.com/r/lyft/envoy-build/tags/

Comment (mattklein123) [inline ci/ci_steps.sh]: (ignore the fancy github text replacement)

Comment (dnoe) [inline ci/ci_steps.sh]: Thanks for the much needed pointer, I was thinking this was what was reported from the "docker images" command which obviously it is not.

Comment (mattklein123) [inline ci/ci_steps.sh]: If there are dev docs that would made sense here please add them.

File: ci/ci_steps.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-ENVOY_BUILD_SHA=fc747b3c2fd49b1260484572071fe4194cd6824d
+ENVOY_BUILD_SHA=1d2a2100708d1012cd61054ce04d4dc4e3b03ff2
```
Comment (dnoe) [inline ci/ci_steps.sh]: This change has no effect outside of this PR until it is merged, correct?

Comment (mattklein123) [inline ci/ci_steps.sh]: Yes, that is correct. But it's failing due to changes in bazel per the other open PR. See the comments there.

**Comment by htuch** [comment]
@dnoe https://github.com/lyft/envoy/pull/1080 now works modulo ASAN and a drop to 97.9 coverage for some reason. Do you want to merge it into your PR and see if it fixes stuff?

**Comment by dnoe** [comment]
Trying it out now.

**Comment by dnoe** [comment]
With @htuch 's change to fix coverage merged I can see an actual coverage report now.  The report is not affected by my change (which only changes behavior under ASAN) but is 97.9%  :( Investigating to see if something else changed.

**Pull request merged**